### ISSUE TITLE
[GroupChat] Add possibility to delete group

### ISSUE THREAD

**Issue by alexandraB99** [status-im/status-desktop#14311]
### Description

When the group chat's owner is leaving the group chat, the chat should be deleted. [Figma](https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?type=design&node-id=10020-353661&mode=dev).

![gc](https://github.com/status-im/status-desktop/assets/31625338/e78d217f-a01e-4780-a273-4b199dd12c6e)

**Comment by jrainville** [comment]
We checked the Discord behaviour, and they do not do this. Instead, when the owner leaves, it promotes a new owner, which I think makes more sense.

@benjthayer @John-44 do we really want the delete group feature? It's weird to implement, because the group should still exist for history at least.
Wouldn't it be better to just make someone else the owner?

**Comment by benjthayer** [comment]
> We checked the Discord behaviour, and they do not do this. Instead, when the owner leaves, it promotes a new owner, which I think makes more sense.
> 
> @benjthayer @John-44 do we really want the delete group feature? It's weird to implement, because the group should still exist for history at least. Wouldn't it be better to just make someone else the owner?

I think we did consider this but it was decided at the time that the nomination process of the new group owner might be tricky - we either add the burden to the existing group chat owner to nominate a new owner and require a flow for nomination and accepting/declining ownerhsip or we run the risk of auto-assigning a member who perhaps shouldn't have that level of control over the group chat.

@jrainville do you know what logic Discord uses to assign a new owner? Is it based on any metrics relating to engagement in the chat etc?

**Comment by jrainville** [comment]
> @jrainville do you know what logic Discord uses to assign a new owner? Is it based on any metrics relating to engagement in the chat etc?

@benjthayer I'm not sure. It seems random. I tried to check in the Discord docs, but couldn't find it.

IMO, just giving the rights to a a random person is not that bad. Being an owner of a group doesn't mean that much. I think one of the only advantage is you can kick people. So if a "bad actor" becomes owner, then I guess he'll kick you, which is not worse than the group just getting deleted if the OG owner leaves.

We should obviously add the option to promote someone as owner. Like right-click on the member in the right side bar to promote.

Now, I don't know that code that well, so maybe it's not that simple to make someone else the owner, considering keys are being handled here. I'm gonna ask the experts here @cammellos @osmaczko

**Comment by cammellos** [comment]
> We should obviously add the option to promote someone as owner. Like right-click on the member in the right side bar to promote.

I think that's how it was before and imo the simplest mental model and the more consonant to the technology, I don't think it was worth the effort to change it at this stage of the app.

Giving admin rights to the earliest member in is not difficult, it requires no protocol changes, by convention people can assume the owner is the oldest member of the chat, so that's not difficult to accommodate (don't think it's worth the development effort as per above).

I would find leaving the chat orphaned fine, deleting the chat on other devices is very poor in terms of UX and I think unexpected, since users might rely on it for storing certain information, and has no security value since it's a cooperative effort.

**Comment by virginiabalducci** [comment]
Hi, I'm looking at the help center on prod https://status.app/help/messaging/delete-a-group-chat and it says that a user can leave and delete a group chat. 
<img width="600" alt="Screenshot 2024-07-19 at 6 02 12 PM" src="https://github.com/user-attachments/assets/96ac891d-f6be-47b3-91d1-22808782409d">

However, on the nightly version from July 17, and the master build up to now, this option only allows to leave the group chat.
![Screenshot 2024-07-19 at 6 01 26 PM](https://github.com/user-attachments/assets/edfbbed4-ac90-4779-8511-eed0b6b6442d)

Since the help page steps have been recently updated, is the "leave and delete group" option to be developed in the future? If this change won't happen soon, I'd suggest updating the instructions on the prod help page to only "Leave group" in order to avoid confusion among the users.

**Comment by jrainville** [comment]
@virginiabalducci I don't think we will implement the "Delete group" feature anytime soon or at all. So I suggest that the docs get updated like you said. cc @jorge-campo

**Comment by jorge-campo** [comment]
> I'd suggest updating the instructions on the prod help page to only "Leave group" in order to avoid confusion among the users.

Thanks, @virginiabalducci – The product has gone through different interactions from design to development, hence the docs mismatch.

@cheny0 , could you please take care of this update? Review the whole document to make sure we aren't missing something. Thanks.

CC/ @jrainville

**Comment by cheny0** [comment]
The article has been updated in this pr: https://github.com/status-im/status-website/pull/1160

**Comment by jrainville** [comment]
Closing as not planned
